### PR TITLE
[WIP] Add progressbar using `FloatProgress` widget for ipynb

### PR DESCRIPTION
**Pull request by rsivapr**
Initial commit for an enhanced progressbar for the ipython notebook. 

Thanks @jsalvatier.

### CODE REVIEW

File: pymc3/progressbar.py [hunk 1]
```diff
@@ -117,10 +107,11 @@ def run_from_ipython():
...
-        else:
-            return TextProgressBar(iters, ipythonprint)
-    else:
+    try:
+        from IPython.html import widgets
+        widgets.FloatProgress()
+        return IPythonNotebookPB(iters)
+    except RuntimeError:
+        return TextProgressBar(iters, ipythonprint)
+    except ImportError:
```
Comment (rsivapr) [inline pymc3/progressbar.py:116]: Don't really like this. @jsalvatier suggested that I move this to head of the file. Still feels very hacky. 
Any thoughts?

**Comment by twiecki** [comment]
Would be great to have this working, thanks! I think @akloster did some work on this so his feedback might also be helpful here.

**Comment by fonnesbeck** [comment]
I'm not sure there is a way around this sort of hack, because IPython never really knows whether it is being controlled from a notebook or a console. This is why IPython notebooks don't display figures inline by default.

File: pymc3/progressbar.py [hunk 2]
```diff
@@ -85,27 +80,22 @@ def ipythonprint(s):
...
-        percentage = int(self.fraction(i))
+        percentage = int(self.percentage(i))
 
-        display(Javascript(
-            "$('div#%s').width('%i%%')" % (self.divid, percentage)))
-        display(Javascript("$('label#%s').text('%i%% in %.1f sec')" %
-                (self.sec_id, fraction, round(elapsed, 1))))
+        # Calculate percent completion, and update progress bar
+        self._widget.value = percentage
+        self._widget.description = ' ({0:>3s}%)'.format('{:d}'.format(percentage))
```
Comment (jsalvatier) [inline pymc3/progressbar.py:98]: Think we also want to record `elapsed` here.

**Comment by jsalvatier** [comment]
@fonnesbeck actually brings up a good point. Couldn't we just look for whatever change `%pylab inline` makes and piggy back on that? 

Also, I think if this is working and has the elapsed seconds we should merge this in.

**Comment by twiecki** [comment]
I think the only way to find the IPython environment is to try and import something and have it fail. A common method is trying to call `get_ipython()`.

**Comment by akloster** [comment]
Sorry for not answering earlier. The problem with using javascript display to run jquery is that it is not necessary and that it accumulates in the notebook output. Which is all sort of weird when rendered without the Kernel running.

I prefer either a custom widget (which can have its own layout, with clock, progress and "stop") or a progressbar/slider widget from the standard widgets. Both solutions come with their own way of communicating.

**Comment by twiecki** [comment]
Thanks @akloster. Accumulating `js` in a cell is pretty bad indeed as it will get saved with the ipynb file.

**Comment by twiecki** [comment]
Closing as this approach causes js code to be dumped to the output.

**Pull request closed without merging**